### PR TITLE
[NFCI][SYCL][Test E2E] Eliminate more warnings

### PR DESCRIPTION
**Pull request by aelovikov-intel**

### CODE REVIEW

**Review by sommerlukas — APPROVED** [review]
Only reviewed the change to the kernel fusion test.

**Review by victor-eds — APPROVED** [review]

File: sycl/test-e2e/Basic/accessor/accessor.cpp [hunk 1]
```diff
@@ -389,7 +389,7 @@ int main() {
             sycl::range<1>{3}, [=](sycl::id<1> index) { dev_acc[index] = 42; });
       });
 
-      auto host_acc = buf.get_access<sycl::access::mode::discard_read_write>();
+      sycl::host_accessor host_acc(buf, sycl::write_only, sycl::no_init);
```
Comment (KseniyaTikhomirova) [inline sycl/test-e2e/Basic/accessor/accessor.cpp:392]: Is access mode changed from read-write to write only intentionally?

**Review by KseniyaTikhomirova — APPROVED** [review]
LGTM

**Comment by aelovikov-intel** [comment]
Failures on

```
Failed Tests (2):
  SYCL :: AtomicRef/accessor.cpp
  SYCL :: DeviceLib/ITTAnnotations/atomic.cpp
```

are unrelated and caused by the known issue in CI where commits used for build/test don't match.

**Pull request merged**